src/index.ts

```diff
@@ -34,73 +38,31 @@ export async function execute(filenames: string[], options: Partial<ExecutionOpt
     const executionOptions = Object.assign(defaultExecutionOptions, options);
 
     let manifest = await PP.getManifest(executionOptions.root);
+    let componentDefinitions = await getComponentDefinitionMap(executionOptions.root);
+
+    let pathFormatter = (component: ComponentDefinition) => {
```

Comment:
Declaring this function inline makes it harder to understand and test.  We should move it out or encapsulate the logic into the interpreter.